src/queryExecutionContext/EndpointComponent/AggregateEndpointComponent.ts

```diff
@@ -10,6 +10,7 @@ export class AggregateEndpointComponent implements ExecutionContext {
   private aggregateValues: any[];
   private aggregateValuesIndex: number;
   private localAggregators: any[];
+  private started: boolean;
```

Comment:
@bchong95 This is the only significant change to logic. I can't quite tell why this changed, but in the new query plan approach, this component returns false initially for `hasMoreResults`. Adding this boolean to track state fixes.

Reply:
I think it's because aggregates are a unique case where you can not return a single document until the source components are fully drained (have no more results). The default behavior for `hasMoreResults` is just whether the source has more results, which is not what you want for aggregates.